### PR TITLE
Fix #3604: Include Aborted Jobs in Garbage Collector Cleanup

### LINKED ISSUE #3604: Garbage Collector is supposed to clean up Aborted volcanojobs
**What happened**: garbagecollector controller does not clean up any Aborted volcanojobs where the `.status.state.phase` is `Aborted`.

**What you expected to happen**: Aborted jobs should be cleaned up.

**How to reproduce it (as minimally and precisely as possible)**:
1. create a volcanojob with `.spec.ttlSecondsAfterFinished` specified.
2. abort this job.
3. `kubectl get vj xxx`, you will see that thee job's `.status.state.phase` is `Aborted`.
4. wait for `ttlSecondsAfterFinished` seconds.
5. however, this job still exists.

**Anything else we need to know?**: The bug seems to occur in the following code. https://github.com/volcano-sh/volcano/blob/v1.10.0-alpha.0/pkg/controllers/garbagecollector/garbagecollector.go#L266

```go
func isJobFinished(job *v1alpha1.Job) bool {
	return job.Status.State.Phase == v1alpha1.Completed ||
		job.Status.State.Phase == v1alpha1.Failed ||
		job.Status.State.Phase == v1alpha1.Terminated
}
```

The condition `job.Status.State.Phase == v1alpha1.Aborted` should not be omitted.

**Environment**:
- Volcano Version: 1.4 to 1.9, up to now

### PR DESCRIPTION
**Pull request by kyooosukedn**
Changes:

 Modified the isJobFinished function in pkg/controllers/garbagecollector/garbagecollector.go to include the Aborted state.

### CODE REVIEW

**Comment by googs1025** [comment]
use `git commit -s ` to handle DCO failed

File: pkg/controllers/garbagecollector/garbagecollector.go [hunk 1]
```diff
@@ -273,6 +273,7 @@ func isJobFinished(job *v1alpha1.Job) bool {
 	return job.Status.State.Phase == v1alpha1.Completed ||
 		job.Status.State.Phase == v1alpha1.Failed ||
 		job.Status.State.Phase == v1alpha1.Terminated
+		job.Status.State.Phase == v1alpha1.Aborted
```
Comment (googs1025) [inline pkg/controllers/garbagecollector/garbagecollector.go:276]: I have a small question: How do we distinguish between "Aborted", "Terminated" and "Failed"? Are these three states the final states of a job?

Comment (Monokaix) [reply pkg/controllers/garbagecollector/garbagecollector.go:276]: That's a good question, `Aborted` is triggered by user, and user can set `Aborted` action in job spec.policy and pods under the pod will be evicted and job can be resumed,  `Terminated` is also set by user but from the code comments that job under this state can not be resumed, while `Failed` is  a state that pods under a job execution failed.
I'm not sure whether whethe a job under `Aborted` state can be resumed, if can be resumed, we should re-consider whether the pr is reasonable: ）

Comment (googs1025) [reply pkg/controllers/garbagecollector/garbagecollector.go:276]: Yes, this is critical, which determines whether it is the final state. If not, it should not be recycled.

Comment (Monokaix) [reply pkg/controllers/garbagecollector/garbagecollector.go:276]: I think this can be close: )

**Comment by Monokaix** [comment]
Thanks for your contribution! You can also find other issues suitable for new users.

**Pull request closed without merging**